cdap-examples/WikipediaPipeline/src/test/java/co/cask/cdap/examples/wikipedia/WikipediaPipelineAppTest.java

```diff
@@ -80,17 +80,17 @@ public void test() throws Exception {
 
     WorkflowManager workflowManager = appManager.getWorkflowManager(WikipediaPipelineWorkflow.NAME);
     // Test with default threshold. Workflow should not proceed beyond first condition.
-    testWorkflow(workflowManager, appConfig, 1);
+    testWorkflow(workflowManager, appConfig, 1, 10);
 
     // Test with a reduced threshold, so the workflow proceeds beyond the first predicate
-    testWorkflow(workflowManager, appConfig, 2, 1);
+    testWorkflow(workflowManager, appConfig, 2, 10, 1);
 
     // Test K-Means
     appConfig = new WikipediaPipelineApp.WikipediaAppConfig("kmeans");
     appRequest = new AppRequest<>(ARTIFACT_SUMMARY, appConfig);
     appManager = deployApplication(APP_ID, appRequest);
     workflowManager = appManager.getWorkflowManager(WikipediaPipelineWorkflow.NAME);
-    testWorkflow(workflowManager, appConfig, 3, 1);
+    testWorkflow(workflowManager, appConfig, 3, 2, 1);
```

Comment:
How come the expected values changed from 10 to 2 here?

Reply:
There is only two inputs in the test, hence the KMeans clustering would at most give 2 clusters. The old code is wrong and is based on the misbehavior in Spark < 2.0 that it always gives the same number of clusters (K) you asked for, regardless of having that many data points or not.